lib/studies/reply/adaptive_nav.dart

```diff
@@ -0,0 +1,716 @@
+import 'dart:math' as math;
+import 'package:flutter/material.dart';
+import 'package:gallery/data/gallery_options.dart';
+import 'package:gallery/l10n/gallery_localizations.dart';
+import 'package:gallery/layout/adaptive.dart';
+import 'package:gallery/studies/reply/colors.dart';
+import 'package:gallery/studies/reply/inbox.dart';
+import 'package:gallery/studies/reply/bottom_drawer.dart';
+
+const _assetsPackage = 'flutter_gallery_assets';
+const _iconAssetLocation = 'reply/icons';
+const _folderIconAssetLocation = '$_iconAssetLocation/twotone_folder.png';
+const double _kFlingVelocity = 2.0;
+
+class AdaptiveNav extends StatefulWidget {
+  const AdaptiveNav({Key key}) : super(key: key);
+
+  @override
+  _AdaptiveNavState createState() => _AdaptiveNavState();
+}
+
+class _AdaptiveNavState extends State<AdaptiveNav> {
+  int _selectedIndex = 0;
+
+  @override
+  Widget build(BuildContext context) {
+    final isDesktop = isDisplayDesktop(context);
+    final isTablet = isDisplaySmallDesktop(context);
+    final localizations = GalleryLocalizations.of(context);
+
+    final _navigationItems = <String, String>{
+      localizations.replyInboxLabel: '$_iconAssetLocation/twotone_inbox.png',
+      localizations.replyStarredLabel: '$_iconAssetLocation/twotone_star.png',
+      localizations.replySentLabel: '$_iconAssetLocation/twotone_send.png',
+      localizations.replyTrashLabel: '$_iconAssetLocation/twotone_delete.png',
+      localizations.replySpamLabel: '$_iconAssetLocation/twotone_error.png',
+      localizations.replyDraftsLabel: '$_iconAssetLocation/twotone_drafts.png',
+    };
+
+    final _folders = <String, String>{
+      'Receipts': _folderIconAssetLocation,
+      'Pine Elementary': _folderIconAssetLocation,
+      'Taxes': _folderIconAssetLocation,
+      'Vacation': _folderIconAssetLocation,
+      'Mortgage': _folderIconAssetLocation,
+      'Freelance': _folderIconAssetLocation,
+    };
+
+    if (isTablet) {
+      return _BuildDesktopNav(
+        selectedIndex: _selectedIndex,
+        extended: false,
+        destinations: _navigationItems,
+        folders: _folders,
+        onItemTapped: _onDestinationSelected,
+      );
+    } else if (isDesktop) {
+      return _BuildDesktopNav(
+        selectedIndex: _selectedIndex,
+        extended: true,
+        destinations: _navigationItems,
+        folders: _folders,
+        onItemTapped: _onDestinationSelected,
+      );
+    } else {
+      return _BuildMobileNav(
+        selectedIndex: _selectedIndex,
+        destinations: _navigationItems,
+        folders: _folders,
+        onItemTapped: _onDestinationSelected,
+      );
+    }
+  }
+
+  void _onDestinationSelected(int index) {
+    setState(() {
+      _selectedIndex = index;
+    });
+  }
+}
+
+class _BuildDesktopNav extends StatefulWidget {
```

Comment:
Nit: Drop `Build` from the name of these widgets and their respective states

```suggestion
class _DesktopNav extends StatefulWidget {
```